### DISCUSSION TITLE
Anybody very familiar with the codebase recommend easiest way to strip it down to one internally sequenced synth for optimal CPU efficiency?

### DISCUSSION THREAD

**Discussion by 01GOD** [copych/AcidBox#29] [Q&A] [answered]
I'd like to add some more hardware and the goal is to keep one synth (and possibly the drum machine) from this project, but remove the second synth and any other non-essential parts of the project for the purpose of making a more CPU efficient edition. 

Anybody very familiar with the codebase recommend easiest way to strip out one of the synths (and possibly the drum machine and external MIDI input) to make a much more CPU-efficient edition with mainly one internally sequenced synth and some effects? 

Thanks in advance!

**Comment by copych** [comment, marked as answer]
There's no just one switch to turn it off. You'll have to comment out everything related to synth2, i.e. Synth2, its buffers, audio_task2, mixer section, AcidBanger section etc.

**Reply by 01GOD** [reply]
> There's no just one switch to turn it off. You'll have to comment out everything related to synth2, i.e. Synth2, its buffers, audio_task2, mixer section, AcidBanger section etc.

Ah. Thanks for the warning about that. By the way, adding the encoder bizarrely seems to have specifically destroyed the functionality of the ESP32's pins I had been using to connect the DAC...and that seems to have happened on 3 ESP32-S3 modules. Don't think that is directly related to AcidBox, but it's fair warning that such issues sadly seem to be possible. If the chinese hotair rework station I had bought hadn't almost melted while holding it, it would be easier to replace those ESP32 modules. Vaguely considering getting another hotair rework station, but that was a scary incident. Fair warning to not buy chinese hotair rework stations also (and to wear safety gear if using any). 

May hook up a logic analyzer later to find out what pin that specifically was causing to cease functioning.

**Comment by copych** [comment]
Esp modules are quite die-hard, so, probably you haven't killed them yet. Re-check the wiring. BTW, most hot-air stations in the market are Chinese,  so pointing particular models would be more useful.

**Reply by 01GOD** [reply]
Tonight used a logic analyzer to confirm all 3 pins (that were connected to the DAC when tried it with that encoder) are now silent on at least one of those ESP32-S3 modules sadly. And that was a new N16R8....sigh...wish it was a cheaper module that passed away. Ordered a replacement N8R2, so hopefully that has enough PSRAM for this project. Either soldering together a new PCB to put it on, or desoldering 32 pin headers to replace it, because no hotair rework station now, so would have to use a hotplate. 

I think I maybe already tossed that hotair rework station. It did have English language instructions written on the control brick, a silver label with black text, a chinese name in latin letters, and a seven segment style red display. 
Also it had the standard black plastic handle common on the lower cost chinese hotair stations. That was the second handle though, so seems probable that was the power control module that was failing. Also been shot with a very hot chinese capacitor before and that sadly left a blister. Bizarrely, some chinese factories idiotically make capacitors without safety valves in the top of them.

**Reply by 01GOD** [reply]
> Esp modules are quite die-hard, so, probably you haven't killed them yet. Re-check the wiring. BTW, most hot-air stations in the market are Chinese, so pointing particular models would be more useful.

A taobao seller failed to send the ESP32 module, so tried making a touch GUI edition again today while waiting for a replacement module to arrive. Posted something today in a new discussion, but it's also pertinent to this discussion as I did strip out the second synth to make enough CPU clock cycles available to run the GUI and synth (and hopefully drum machine also) at full speed. 

Here is a link to that new discussion: 
https://github.com/copych/AcidBox/discussions/30

**Comment by 01GOD** [comment]
I tried to trudge through and chop through every compiler error along the way to deal with that Arduino IDE compiler mess, but it was such a disaster when it couldn't find general.ino or general.c or config.h and then general.c couldn't find AcidBox.ino and so on. ALL of them open automatically when AcidBox.ino is opened, so the IDE does know they are supposed to be linked together, but maybe that IDE is maybe trying to synthesize its own VERY WRONG .h files or similar. Hasn't seemed so easy to majorly restructure a relatively large codebase I am not so familiar with.
